client/consensus/grandpa/interfaces.go

```diff
@@ -26,13 +26,19 @@ type ForkTree[H comparable, N constraints.Unsigned] interface {
 	PendingChanges() []PendingChange[H, N]
 }
 
-// PublicKey interface
-type PublicKey interface {
-	Verify(msg, sig []byte) (bool, error)
-	Encode() []byte
-	Decode([]byte) error
-	Hex() string
-}
+type Telemetry interface{}
 
-type Telemetry interface {
+// AuxStore is part of the substrate backend.
+// Provides access to an auxiliary database.
+//
+// This is a simple global database not aware of forks. Can be used for storing auxiliary
+// information like total block weight/difficulty for fork resolution purposes as a common use
+// case.
+type AuxStore interface {
+	// Insert auxiliary data into key-Value store.
+	//
+	// Deletions occur after insertions.
+	Insert(insert []databaseKeyValue, deleted [][]byte) error
```

Comment:
You shouldn't use private types in public interface methods.  It may be easier to read to create a `KeyValue` struct.